### ISSUE TITLE
Tab Trackers: percentage and number apps are incorrect

### ISSUE THREAD

**Issue by Jean-BaptisteC** [Exodus-Privacy/exodus-android-app#103]
Issue:
- Device: Pixel 6
- OS: Android 12
- EP Version: Preview

How reproduce:

- Go trackers tab 
- Look percentage and apps number
- Consult trackers
- Compare informations on trackers tab and trackers
- Informations are different

Exemple:

- Google Firebase Analytics
- On trackers tab
- 45% | 29 apps
- On trackers
- 39% | 25 apps

**Comment by Jean-BaptisteC** [comment]
| Version | Detected |
| --- | --- |
| Android 5 | Yes |
| Android 7 | Yes |
| Android 8 | Yes |
| Android 12 | Yes |

**Comment by ItsIgnacioPortal** [comment]
Verified on v2.2.1, Android 9. At first I though it was a rounding difference, but then I checked my most "popular" tracker, Google Firebase Analytics, and in the Trackers page it says 33%, but in it's individual tracker page it says 22%. No rounding error can do that.

**Comment by ItsIgnacioPortal** [comment]
Ok, so I've investigated this.

# Individual page

Searching `Present in`, we find an XML with strings referencing `tracker_presence`. Searching `tracker_presence` we find:
> exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/TrackerFragment.java

```java
String presence = context.getResources().getString(R.string.tracker_presence, displayedApps;
```

But the actual text is set in line 144:
```java
trackerBinding.trackerPresenceNb.setText(String.format("%s%%", percent));
```

`percent` is calculated in line 134
```java
int percent = displayedApps * 100 / total;
```

```java
int displayedApps = appListFragment.getDisplayedApps();
```

displayedApps is modified by filter() in: 
> exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/adapters/ApplicationListAdapter.java
```java
public void filter(AppListFragment.Type type, Object filterObject) {
	displayedApp = 0;
	if (type.equals(AppListFragment.Type.NAME)) {
		filter = filterObject;
		filterType = type;
		String filterStr = (String) filterObject;

		Pattern p = Pattern.compile(Pattern.quote(filterStr.trim()), Pattern.CASE_INSENSITIVE);
		for (ApplicationViewModel app : applicationViewModels) {
			app.isVisible = p.matcher(app.label).find();
			if (app.isVisible)
				displayedApp++;
		}
	} else if (type.equals(AppListFragment.Type.TRACKER)) {
		filter = filterObject;
		filterType = type;
		Long filterLng = (Long) filterObject;

		for (ApplicationViewModel app : applicationViewModels) {
			app.isVisible = false;
			if (app.trackers != null) {
				for (Tracker tracker : app.trackers) {
					if (tracker.id == filterLng) {
						app.isVisible = true;
						displayedApp++;
						break;
					}
				}
			}
		}
	}
	notifyDataSetChanged();
}
```


going back, `total` is set in:
> exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/TrackerFragment.java
```java
int total = appListFragment.getTotalApps();
```

`getTotalApps` is defined in:
> exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/AppListFragment.java 
```java
    public int getTotalApps() {
        if (adapter == null)
            return 0;
        return adapter.getItemCount();
    }
```

`adapter.getItemCount()` was defined in:
> exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/adapters/ApplicationListAdapter.java
```java
public int getItemCount() {
    return applicationViewModels.size();
}
```


# General tracker page

My best guess is that `exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/MyTrackersFragment.java` is the general trackers page, because it seems that "fragments" are how you call pages on Android.

The page is synchronized on line 132
```java
MyTrackersListAdapter myTrackersListAdapter = new MyTrackersListAdapter(myTrackers, MyTrackersFragment.this, finalMaxValue, appInstalled);
trackerBinding.trackers.setAdapter(myTrackersListAdapter);
``` 

Investigating this `MyTrackersListAdapter` class:
> exodus-android-app/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/adapters/MyTrackersListAdapter.java

Here we can CTRL+F `percent` and we find:
```java
float percent = (float) myTracker.number / (float) max;
int percentApp = myTracker.number * 100 / installedApps;
```

from looking at line 66 we can conclude that the actual percentage being displayed is `percentApp`:
```java
holder.viewDataBinding.percentVal.setText(String.format("%s %%", percentApp));
```

# Theory

The formulas for `percent` and `percentApp` are very similar:
```java
int percent = displayedApps * 100 / total;
int percentApp = myTracker.number * 100 / installedApps;
```

So, there are two options:
- `total` and `installedApps` are different
- `displayedApps` and `myTracker.number` are different

If I actually knew how to make Android apps in Java, I would now run the app in debug mode and make a toast that showed the value of these variables. But since I don't, this is as much as I can help right now 😅

**Comment by Jean-BaptisteC** [comment]
Thanks for your diagnostic,
But i keep open this issue and all issues present in column Tests to check if bug is fixed on 3.0 Version;

**Comment by Jean-BaptisteC** [comment]
Problem is fixed on version 3.0 debug
- Percentage and nb apps on trackers fragments = percentage and nb apps on trackers details fragments =>OK
